includes/User.php

```diff
@@ -2123,8 +2123,6 @@ private function clearSharedCache() {
 		if( $this->mId ) {
 			global $wgMemc, $wgSharedDB; # Wikia
 			$wgMemc->delete( $this->getCacheKey() );
-			// Wikia: and save updated user data in the cache to avoid memcache miss and DB query
-			$this->saveToCache();
```

Comment:
This is a weird and unnecessary side effect. The function is called "clearSharedCache", not "clearSharedCache-thenResaveSharedCache".

It also has the annoying consequence of sometimes saving stale data when called from `User::invalidateCache`.  The way this happens is first we create a user object so we can clear its cache:
```php
$myUser = User::newFromId( 1 );
```

As that user object is being created, it tries to pull data for itself out of cache [here](https://github.com/Wikia/app/blob/dev/includes/User.php#L356). Chances are it succeeds and now we have a user object initialized from cache.

We then call `$myUser->invalidateCache()`. That method creates a new value for the member variable `mTouched`  [here](https://github.com/Wikia/app/blob/dev/includes/User.php#L2153), which is used to indicate the last time this user was updated, and then saves that value to the db. So far so good.

The problem is `User::invalidateCache()` would call `User::clearSharedCache` which would call `User::saveToCache` which ends up saving the data we pulled from cache originally, but with the new value for mTouched! So now we've saved old data, with the latest mTouched value and stale cache is served.

Let's just get rid this.